contracts/core/src/contract.rs

```
@@ -371,6 +375,32 @@ fn execute_set_non_native_rewards_receivers(
     ))
 }
 
+fn execute_update_withdrawn_amount(
+    deps: DepsMut<NeutronQuery>,
+    _env: Env,
+    info: MessageInfo,
+    batch_id: u128,
+    withdrawn_amount: Uint128,
+) -> ContractResult<Response<NeutronMsg>> {
+    let config = CONFIG.load(deps.storage)?;
+    if info.sender != config.withdrawal_manager_contract {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let mut batch = unbond_batches_map().load(deps.storage, batch_id)?;
+    if batch.status == UnbondBatchStatus::Withdrawn {
+        batch.withdrawn_amount =
+            Some(batch.withdrawn_amount.unwrap_or_default() + withdrawn_amount);
+    }
```

I would rather exit with an error instead of silently doing nothing